app/controllers/users_controller.rb

```diff
@@ -0,0 +1,73 @@
+class UsersController < ApplicationController
+
+  before_action :account_views, only: [:show, :edit]
+
+  def index
+    @users = User.all
+  end
+
+  def show
+    @user = User.find_by(id: params[:id])
+  end
+
+  def create
+    auth_hash = request.env['omniauth.auth']
+    if auth_hash['uid']
+      @user = User.find_by(uid: auth_hash[:uid], provider: 'github')
+      if @user.nil?
+        @user = User.build_from_github(auth_hash)
+        successful_save = @user.save
+        if successful_save
+          flash[:status] = :success
+          flash[:result_text] = "#{@user.name} logged in successfully"
+          session[:uid] = @user.uid
+          session[:user_id] = @user.id
```

Comment:
There's a lot of setting on session the values uid and user_id, and from what I can tell, it's mostly that the code didn't conform to using one or the other, even though they represent the same thing. You'll prevent bugs and confusion if you conform to one standard here!